kaldb/src/main/java/com/slack/kaldb/server/KaldbQueryServiceBase.java

```diff
@@ -19,23 +18,13 @@ public void search(
     LOG.info(
         String.format("Search request received: '%s'", request.toString().replace("\n", ", ")));
 
-    // There is a nuance between handle vs handleAsync/whenCompleteAsync
-    // handleAsync/whenCompleteAsync will cause the callback to be invoked from Java's default
-    // fork-join pool
-    doSearch(request)
-        .handle(
-            (result, t) -> {
-              if (t != null) {
-                LOG.error("Error completing the future", t);
-                responseObserver.onError(t);
-              } else {
-                responseObserver.onNext(result);
-                responseObserver.onCompleted();
-              }
-              return null;
-            });
+    try {
+      responseObserver.onNext(doSearch(request));
+    } catch (Exception e) {
+      responseObserver.onError(e);
+    }
+    responseObserver.onCompleted();
```

Comment:
Should we still call `onCompleted` when an error occurs? It appears previously `onCompleted` only executed when there was no error.

Reply:
done